syntax = "proto3";

import "google/protobuf/empty.proto";
import "google/protobuf/wrappers.proto";

package task;

// Task 메시지 정의
message Task {
  int32 id = 1; // 고유 식별자
  string content = 2; // 할일 내용
  string due_date = 3; // 마감일 (문자열 형식, 예: "2023-12-31")
  bool completed = 4; // 완료 여부
  int32 prior_rank = 5; // 우선 순위
}

// 할일 관련 서비스 정의
service TaskService {
  // 할일 추가
  rpc CreateTask(Task) returns (Task);
  // 할일 목록 조회
  rpc ListTasks(google.protobuf.Empty) returns (stream Task);
  // 특정 할일 조회
  rpc GetTask(google.protobuf.Int32Value) returns (Task);
  // 할일 업데이트
  rpc UpdateTask(Task) returns (Task);
  // 할일 삭제
  rpc DeleteTask(google.protobuf.Int32Value) returns (google.protobuf.Empty);
}
